### PR TITLE
Update documentation for the event feature

### PR DESCRIPTION
**Pull request by andrewtanJS**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -261,45 +261,50 @@ image::UML REST API.png[width="400"]
...
-=== Schedule feature
+//tag::event[]
+=== Event feature
 ==== Current Implementation
 
-The schedule command is a completely new feature that allows the user to schedule events within ConTAct.
-It follows a similar command flow as that of the address book, but it will have it's own separate model and
-.xml file for storage. Furthermore, to facilitate this command, a new data type is introduced: the Event class.
-This Event class has it's own unique fields that the user can specify to schedule an event:
+Th event feature is largely centered around the `Event` class, and is integrated into ConTAct in a similar fashion to the address book.
```
Comment (alyip98) [inline docs/DeveloperGuide.adoc]: Th

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -261,45 +261,50 @@ image::UML REST API.png[width="400"]
...
+The main commands supporting this feature is `schedule` and `cancel`, which both interact with the `Calendar` model.
```
Comment (alyip98) [inline docs/DeveloperGuide.adoc]: commands supporting this feature ~~is~~ are

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (alyip98) [reply docs/DeveloperGuide.adoc]: perhaps also include a brief mention how schedule and cancel interact with the Calendar model -- i.e. adding and deleting Events

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -261,45 +261,50 @@ image::UML REST API.png[width="400"]
...
+The `undo/redo` and `clear` commands were also modified to extend towards the calendar model.
+
+The Event class has these fields that the user can specify to schedule an event:
 
 * Event Name: The name of the event, valid if it is not an empty string.
 * Date: The date of the event, valid if expressed in this forms:  `DD-MM-YYYY`.
```
Comment (alyip98) [inline docs/DeveloperGuide.adoc]: form~~s~~

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -261,45 +261,50 @@ image::UML REST API.png[width="400"]
...
+This list is exposed to the UI to display the list accordingly.
+Furthermore, the event data is permanently stored in a `calendar.xml` file for use between sessions.
+
+Below are several commands that allow the user to interact with the events:
 
-Currently, the main operation is the `schedule` command, and it is implemented as such:
+===== Schedule
 
-* `schedule event/EVENT_NAME date/DATE start/START_TIME end/END_TIME descr/DESCRIPTION`
+The `schedule` command allows the user to add a unique event to the calendar, and it is implemented as such:
```
Comment (alyip98) [inline docs/DeveloperGuide.adoc]: what does "unique" refer to here, which fields are checked for uniqueness?

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -261,45 +261,50 @@ image::UML REST API.png[width="400"]
...
 
-This will simply create an event with the specified fields and store it similar to how the addressbook stores a student.
-Furthermore, the schedule command ensures that all the fields specified must be valid for it to be a success.
+This will simply create an event with the specified fields and store it accordingly.
 
-Once scheduled, the list of events will be sorted by chronological order (taking the date and the start time).
+Once scheduled, the events will be inserted in its sorted position (by chronological order).
 
 The GUI itself will have a calendar component which will display the user's current list of events that they have
 scheduled, while adjusting for the current date.
```
Comment (alyip98) [inline docs/DeveloperGuide.adoc]: what does "adjusting for the current date" mean? does it automatically scroll to show the first upcoming event

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -312,16 +317,58 @@ The diagram below shows how the schedule command is parsed in the Logic componen
...
+Also, the interactions between the components work in much the same way as the schedule command.
+
+===== Undo/Redo
+
+The `undo` and `redo` command was extended to include the actions made by the user while interacting with the calendar.
+
+The undo/redo feature is also facilitated by a VersionedCalendar, and further details are as described in the undo/redo feature.
+
+The most pertinent aspect of this feature is how the model component handles the undo/redo. The ModelManager simply keeps track of _which_
+model -- address book or calendar -- had committed a change, and handles it accordingly whenever the user wishes to undo or redo.
```
Comment (alyip98) [inline docs/DeveloperGuide.adoc]: perhaps mention what data structure/method is used to track the combined history

File: docs/DeveloperGuide.adoc [hunk 7]
```diff
@@ -963,6 +1012,34 @@ Use case ends.
...
+*MSS*
+
+1. User cancels an event with the specified details
+2. ConTAct confirms the deletion of said event
++
+Use case ends.
+
+*Extensions*
+
+[noe]
```
Comment (alyip98) [inline docs/DeveloperGuide.adoc]: typo

File: docs/UserGuide.adoc [hunk 8]
```diff
@@ -127,20 +127,35 @@ Marks the attendance of the student with index number 1 as present.
...
+=== Schedule events: `schedule`
 
 Schedules an event for the user to keep track of. +
-Format: `schedule event/EVENT_NAME date/DATE start/TIME_START end/TIME_END descr/DESCRIPTION`
+Format: `schedule event/EVENT_NAME date/DATE start/TIME_START end/TIME_END [descr/DESCRIPTION]`
 
 ****
-* The event will be scheduled and reflected on the user's list of events
-* Invalid dates and times will not be scheduled e.g. `32-12-2018` will not schedule any event. Also, the year range is between 1600 and 9999
+* The event will be scheduled and reflected on the user's list of events.
```
Comment (alyip98) [inline docs/UserGuide.adoc]: perhaps briefly mention the uniqueness criteria here, in case for some reason the user wants to create 2 identical events

**Comment by andrewtanJS** [comment]
Sure thing, thanks for the review. Will update soon

File: docs/DeveloperGuide.adoc (hunk 2 above)
Comment (andrewtanJS) [reply docs/DeveloperGuide.adoc]: fixed typo, thanks!

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (andrewtanJS) [reply docs/DeveloperGuide.adoc]: aight, added a small section

File: docs/DeveloperGuide.adoc (hunk 6 above)
Comment (andrewtanJS) [reply docs/DeveloperGuide.adoc]: Added a section detailing this.

File: docs/DeveloperGuide.adoc (hunk 7 above)
Comment (andrewtanJS) [reply docs/DeveloperGuide.adoc]: fixed.

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (andrewtanJS) [reply docs/DeveloperGuide.adoc]: fixed

File: docs/UserGuide.adoc (hunk 8 above)
Comment (andrewtanJS) [reply docs/UserGuide.adoc]: Added a new point

**Pull request merged**